api/src/main/java/io/onetable/client/HudiSourceConfig.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+ 
+package io.onetable.client;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Locale;
+
+import lombok.Value;
+
+import io.onetable.model.schema.PartitionTransformType;
+import io.onetable.spi.extractor.SourcePartitionSpecExtractor;
+
+public interface HudiSourceConfig {
+  public String getPartitionSpecExtractorClass();
+
+  List<PartitionFieldSpec> getPartitionFieldSpecs();
+
+  @Value
+  static class PartitionFieldSpec {
+    String sourceFieldPath;
+    PartitionTransformType transformType;
+    String format;
+  }
+
+  static List<PartitionFieldSpec> parsePartitionFieldSpecs(String input) {
```

Comment:
This method is not invoked by any class. The `HudiSourceImpl` class has its own private implementation. It is doubtful whether this method should be part of the interface contract.